src/http/one/TeChunkedParser.cc

```diff
@@ -121,7 +126,7 @@ Http::One::TeChunkedParser::parseChunkMetadataSuffix(Tokenizer &tok)
     // bad or insufficient input, like in the code below. TODO: Expand up.
     try {
         parseChunkExtensions(tok); // a possibly empty chunk-ext list
-        skipLineTerminator(tok);
+        tok.skipRequired("CRLF after [chunk-ext]", Http1::CrLf());
```

Comment:
> please check the functionality-tests is correct

Done. I believe our CI tests revealed a bug in PR changes. You should be able to trigger the same bug using pretty much any basic chunked-encoded message. FWIW, the failing test was using a 15-byte (before encoding) chunk-encoded body like this:

```
f\r\n
body-bytes-here\r\n
```

The _combination_ of old parseChunkExtensions() and new tok.skipRequired() does not work because parseChunkExtensions() consumes any BWS. In case of a simple canonical chunk, parseChunkExtensions() consumes the leading CR in the chunk-terminating CRLF (that PR code now _requires_ "after [chunk-ext]"). That innocent CR looks like BWS to Http::One::ParseBws().

To fix this, we should fix parseChunkExtensions() (at least) to consume nothing but chunk extensions. Those extensions may include BWS, of course, but if we want to be strict in its caller, then parseChunkExtensions()[^1] cannot consume BWS that is _not_ a part of a chunk extension. 

The same restrictions apply to any code that goes before the new tok.skipRequired() calls -- that code can no longer consume CR from the terminating CRLF.

[^1]: parseChunkExtensions() itself and the code parseChunkExtensions() calls, of course.

Reply:
Ah. Thank you. I have adjusted the parser in [4a39435](https://github.com/squid-cache/squid/pull/1498/commits/4a39435cf8568fabae2bbb8c84d753569bf04918) to also be more strict about the BWS it consumes from those two places, no longer skipping the relaxed whitespace characters should meet the need without re-opening the bugs handling IBM server and ICAP responses.